pkgs/applications/audio/ardour/default.nix

```diff
@@ -127,21 +143,22 @@ in stdenv.mkDerivation rec {
   # be available. Since this is an unzipped tarball fetched from github we
   # have to do that ourself.
   postPatch = ''
-    printf '#include "libs/ardour/ardour/revision.h"\nnamespace ARDOUR { const char* revision = \"${tag}-${builtins.substring 0 8 src.rev}\"; }\n' > libs/ardour/revision.cc
+    printf '#include "libs/ardour/ardour/revision.h"\nnamespace ARDOUR { const char* revision = "6.0"; const char* date = "2020-05-23"; }\n' > libs/ardour/revision.cc
+    sed 's|/usr/include/libintl.h|${glibc.dev}/include/libintl.h|' -i wscript
     patchShebangs ./tools/
+    sed "382 s/mingw/x86_64/" -i libs/ardour/wscript
```

Comment:
This is a bit of a hack to get it to compile.
Open to suggestions!

Reply:
A better solution would be an actual patch. This is too brittle, because (IIRC) a failed `sed` won't halt the build. A failed patch will.

Reply:
To make a proper patch I need to know what to change this code to.
I tried removing the if statement altogether, but got errors about unexpected indentation on line 383.
Tried to remove the indentation but kept getting the same error.
ATM I'm just changing the condition, but that only works when building for x86_64.

Reply:
Well, you're already changing that part from `mingw` to `x86_64` -- why not a patch that does the same? This way, git will be able to automatically resolve slight location changes (e.g. added line here or there), but will fail if the patch doesn't apply (e.g. that line gets removed entirely).

Reply:
This could go upstream, but I'd first like to understand why it's needed for NixOS only.
other distros with the  GNU assembler  2.31.1 don't have this issue.

Reply:
@cole-h That "fix" only works when building for x86_64.  Ardour should run on many platforms.

Reply:
A failed `sed` does not halt the build. I changed this to `sed "382 s/fdjsaklfjldska/x86_64/" -i libs/ardour/wscript` and it continued building just fine (at least, until it errored out due to what this is trying to fix).

`echo asdf | sed "s/fdsa/jkl;" && echo $?` -> `asdf` and `0`.

But yes, tracking down why this is necessary is a better idea.

Reply:
I guess the reason why definition usually work for waf is because they pick up "gcc" as the entry point: https://gitlab.com/ita1024/waf/-/blob/2e9954e08736e6cfa2aa721073f11f69c9bd5203/waflib/Tools/gas.py#L14
My guesses for what can be special with nixpkgs is
1. Something sets `AS=as`
2. cc-wrapper does something special (I can't see it looking at the code, though)

Reply:
While it would be nice to find the root cause, I think for now the patch provided by @prusnak is a good solution.